ios/NotificationServiceExtension/NotificationService.swift

```diff
@@ -0,0 +1,236 @@
+//
+//  NotificationService.swift
+//  NotificationServiceExtension
+//
+//  Created by Andrew Rosiclair on 12/8/23.
+//
+
+import AirshipServiceExtension
+import os.log
+import Intents
+
+class NotificationService: UANotificationServiceExtension {
+  
+  var contentHandler: ((UNNotificationContent) -> Void)?
+  var bestAttemptContent: UNMutableNotificationContent?
+  let log = OSLog(subsystem: Bundle.main.bundleIdentifier ?? "com.expensify.chat.dev.NotificationServiceExtension", category: "NotificationService")
+  
+  override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+    os_log("[NotificationService] didReceive() - received notification", log: log)
+    
+    self.contentHandler = contentHandler
+    guard let bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent) else {
+      contentHandler(request.content)
+      return
+    }
+    
+    if #available(iOSApplicationExtension 15.0, *) {
+      configureCommunicationNotification(notificationContent: bestAttemptContent, contentHandler: contentHandler)
+    } else {
+      contentHandler(bestAttemptContent)
+    }
+  }
+  
+  @available(iOSApplicationExtension 15.0, *)
+  func configureCommunicationNotification(notificationContent: UNMutableNotificationContent, contentHandler: @escaping (UNNotificationContent) -> Void) {
+    var notificationData: NotificationData
+    do {
+      notificationData = try parsePayload(notificationContent: notificationContent)
+    } catch ExpError.runtimeError(let errorMessage) {
+      os_log("[NotificationService] configureCommunicationNotification() - couldn't parse the payload '%@'", log: log, type: .error, errorMessage)
+      contentHandler(notificationContent)
+      return
+    } catch {
+      os_log("[NotificationService] configureCommunicationNotification() - unexpected error while parsing payload", log: log, type: .error)
+      contentHandler(notificationContent)
+      return
+    }
+    
+    // Create an intent for the incoming message
+    let intent: INSendMessageIntent = createMessageIntent(notificationData: notificationData)
+
+    // Use the intent to initialize the interaction.
+    let interaction = INInteraction(intent: intent, response: nil)
+
+
+    // Interaction direction is incoming because the user is
+    // receiving this message.
+    interaction.direction = .incoming
+
+
+    // Donate the interaction before updating notification content.
+    interaction.donate { error in
+        if error != nil {
+            os_log("[NotificationService] configureCommunicationNotification() - failed to donate the message intent", log: self.log, type: .error)
```

Comment:
```suggestion
            os_log("[NotificationService] configureCommunicationNotification() - failed to donate the message intent", log: self.log, type: .error, error.localizedDescriptio)
```

I am not sure how we do swift error logging here.

Reply:
I'm also new to logging with Swift. What would this change do?

Reply:
It would print a more reader friendly error. I can see it's being used in our Pods but not in the project itself. We can ignore this I think then.